piecrust/src/session.rs

```diff
@@ -49,16 +49,44 @@ unsafe impl Sync for Session {}
 /// then be [`commit`]ed to, or discarded by simply allowing the session to
 /// drop.
 ///
-/// New modules are to be `deploy`ed in the context of a session. Metadata
-/// queryable by modules can be set using [`set_meta`].
+/// New modules are to be `deploy`ed in the context of a session.
 ///
 /// [`VM`]: crate::VM
 /// [`queried`]: Session::query
 /// [`transacted`]: Session::transact
 /// [`commit`]: Session::commit
-/// [`set_meta`]: Session::set_meta
 #[derive(Debug)]
 pub struct Session {
+    inner: &'static mut SessionInner,
```

Comment:
Why not make inner `Rc<RefCell<SessionInner` and let Rust handle the ref count?

On the consuming side, session would expose `borrow()` and `borrow_mut`, or is that not granular enough for this situation?

Reply:
Take a look now and tell me if it makes more sense.